packages/jest-config/src/__tests__/normalize.test.ts

```diff
@@ -2204,3 +2204,110 @@ describe('runInBand', () => {
     expect(options.runInBand).toBe(true);
   });
 });
+
+describe('coverageProvider', () => {
+  let requireResolveSpy: any;
+  const consoleSpy = jest.spyOn(console, 'warn');
+
+  beforeEach(() => {
+    consoleSpy.mockImplementation(() => {});
+  });
+
+  afterEach(() => {
+    consoleSpy.mockReset();
+    requireResolveSpy?.mockRestore();
+  });
+
+  test('should accept v8 without babel-plugin-istanbul', async () => {
+    const {options} = await normalize(
+      {
+        coverageProvider: 'v8',
+        rootDir: '/root/',
+      },
+      {} as Config.Argv,
+    );
+    expect(options.coverageProvider).toBe('v8');
+  });
+
+  test('should work with babel when babel-plugin-istanbul is available', async () => {
+    // Mock require.resolve to simulate babel-plugin-istanbul being available
+    const originalResolve = require.resolve;
+    requireResolveSpy = jest.spyOn(require, 'resolve');
+    requireResolveSpy.mockImplementation((id: string) => {
+      if (id === 'babel-plugin-istanbul') {
+        return '/fake/path/to/babel-plugin-istanbul';
+      }
+      return originalResolve(id);
+    });
+
+    const {options} = await normalize(
+      {
+        coverageProvider: 'babel',
+        rootDir: '/root/',
+      },
+      {} as Config.Argv,
+    );
+    expect(options.coverageProvider).toBe('babel');
+    expect(consoleSpy).not.toHaveBeenCalled();
+  });
+
+  test('should warn and fallback to v8 when babel is requested but babel-plugin-istanbul is missing', async () => {
+    // Use doMock to override babel-plugin-istanbul module resolution at the Jest level
+    jest.doMock('babel-plugin-istanbul', () => {
+      throw new Error('Module not found');
+    });
+
+    // Force re-require of normalize to pick up the mocked babel-plugin-istanbul
+    jest.resetModules();
+    const normalizeModule = require('../normalize');
```

Comment:
Can we use dynamic import here?

Reply:
Where is the dynamic import? Should it be `const normalizeModule = await import('../normalize');` ?